website/docs/r/mssql_virtual_machine.html.markdown

```diff
@@ -0,0 +1,220 @@
+  allocation_method   = "Dynamic"
+}
+
+resource "azurerm_network_security_group" "nsg" {
+  name                = "examplensg"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_network_security_rule" "RDPRule" {
+  name                        = "RDPRule"
+  resource_group_name         = azurerm_resource_group.example.name
+  priority                    = 1000
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = 3389
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.nsg.name
+}
+
+resource "azurerm_network_security_rule" "MSSQLRule" {
+  name                        = "MSSQLRule"
+  resource_group_name         = azurerm_resource_group.example.name
+  priority                    = 1001
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = 1433
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.nsg.name
+}
+
+resource "azurerm_network_interface" "example" {
+  name                      = "exampleni"
+  location                  = azurerm_resource_group.example.location
+  resource_group_name       = azurerm_resource_group.example.name
+  network_security_group_id = azurerm_network_security_group.nsg.id
+
+  ip_configuration {
+    name                          = "exampleconfiguration1"
+    subnet_id                     = azurerm_subnet.example.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.vm.id
+  }
+}
+
+resource "azurerm_virtual_machine" "example" {
+  name                  = "examplevm"
+  location              = azurerm_resource_group.example.location
+  resource_group_name   = azurerm_resource_group.example.name
+  network_interface_ids = [azurerm_network_interface.example.id]
+  vm_size               = "Standard_DS13"
+
+  storage_image_reference {
+    publisher = "MicrosoftSQLServer"
+    offer     = "SQL2019-WS2019"
+    sku       = "SQLDEV"
+    version   = "laexample"
+  }
+
+  storage_os_disk {
+    name          = "exampleOSDisk"
+    vhd_uri       = "${azurerm_storage_account.example.primary_blob_endpoint}vhds/exampleOSDisk.vhd"
+    caching       = "ReadOnly"
+    create_option = "FromImage"
+  }
+
+  os_profile {
+    computer_name  = "winhost01"
+    admin_username = "exampleadmin"
+    admin_password = "Password1234!"
+  }
+
+  os_profile_windows_config {
+    timezone                  = "Pacific Standard Time"
+    provision_vm_agent        = true
+    enable_automatic_upgrades = true
+  }
+}
```

Comment:
Since the creation of sql vm has some exigences of the vm providing , such as "storage_image_reference", enable "provision_vm_agent",network rules and so on, I come up with a concern that the user might not be able to create the sql vm without the steps.

Reply:
I think it might be better to out a complete example into the examples folder and shorten this to just the resource given how long it is here..